react-accordion-component/my-app/src/AccordionItem.tsx

```diff
@@ -0,0 +1,25 @@
+import { HiChevronDown, HiChevronRight } from 'react-icons/hi';
+import { Topic } from './Accordion';
+
+type Props = {
+  topic: Topic;
+  index: number;
+  currentIndex: number | null;
+  onSelect: (index: number) => void;
```

Comment:
An alternative to passing the index around is to just pass `topic` and an `isOpen` flag:
```
type Props = {
  topic: Topic;
  isOpen: boolean;
  onSelect: (topic: Topic) => void;
}
...
onClick={() => onSelect(topic)}
// When calling AccordionItem:
<AccordionItem topic={topic} isOpen={currentTopic === topic} />
```

Functionally, the two approaches are essentially the same, but I like how things simplify with this new approach.